saleor/graphql/discount/types/promotions.py

```diff
@@ -99,3 +115,65 @@ def resolve_promotion(root: models.PromotionRule, info: ResolveInfo):
     @staticmethod
     def resolve_channels(root: models.PromotionRule, info: ResolveInfo):
         return ChannelsByPromotionRuleIdLoader(info.context).load(root.id)
+
+
+class PromotionEvent(ModelObjectType[models.PromotionEvent]):
+    id = graphene.GlobalID()
+    date = graphene.DateTime(description="Date when event happened.")
+    type = PromotionEventsEnum(description="Promotion event type.")
+    user = graphene.Field(User, description="User who performed the action.")
+    app = graphene.Field(App, description="App that performed the action.")
+    message = graphene.String(description="Content of the event.")
+    rule_id = graphene.ID(description="ID of the rule.")
+
+    class Meta:
+        description = "History log of the promotion." + ADDED_IN_315 + PREVIEW_FEATURE
+        interfaces = [relay.Node]
+        model = models.PromotionEvent
+        doc_category = DOC_CATEGORY_DISCOUNTS
+
+    @staticmethod
+    def resolve_user(root: models.PromotionEvent, info):
+        def _resolve_user(user):
+            requester = get_user_or_app_from_context(info.context)
+            if not requester:
+                return None
+            check_is_owner_or_has_one_of_perms(
+                requester,
+                user,
+                AccountPermissions.MANAGE_USERS,
```

Comment:
Shouldn't we validate here only `MANAGE_STAFF`? Is it possible that the `user` will not be a `Staff`?